contracts/Assertion.sol

```diff
@@ -1,38 +1,24 @@
 // SPDX-License-Identifier: MIT
 
-pragma solidity ^0.8.4;
+pragma solidity ^0.8.16;
 
-import {Hub} from "./Hub.sol";
 import {AssertionStorage} from "./storage/AssertionStorage.sol";
+import {ContractStatus} from "./abstract/ContractStatus.sol";
+import {Initializable} from "./interface/Initializable.sol";
 import {Named} from "./interface/Named.sol";
 import {Versioned} from "./interface/Versioned.sol";
 import {AssertionStructs} from "./structs/AssertionStructs.sol";
 
-contract Assertion is Named, Versioned {
+contract Assertion is Named, Versioned, ContractStatus, Initializable {
     event AssertionCreated(bytes32 indexed assertionId, uint128 size, uint32 triplesNumber, uint96 chunksNumber);
 
     string private constant _NAME = "Assertion";
-    string private constant _VERSION = "1.0.0";
+    string private constant _VERSION = "1.0.1";
 
-    Hub public hub;
     AssertionStorage public assertionStorage;
 
-    constructor(address hubAddress) {
-        require(hubAddress != address(0), "Hub Address cannot be 0x0");
-
-        hub = Hub(hubAddress);
-        initialize();
-    }
-
-    modifier onlyHubOwner() {
-        _checkHubOwner();
-        _;
-    }
-
-    modifier onlyContracts() {
-        _checkHub();
-        _;
-    }
+    // solhint-disable-next-line no-empty-blocks
+    constructor(address hubAddress) ContractStatus(hubAddress) {}
```

Comment:
Why did we remove calling initialize method in constructor, shouldn't it be there to avoid additional blockchain transactions - calling initialize?

Reply:
Because we moved setting in the Hub, initialize() etc. to the end of the deployment and it will be done in 1 tx. With initialize() in the constructor, we will have dependencies issue which can result in reverted deployment
<img width="785" alt="image" src="https://user-images.githubusercontent.com/71610423/230346819-bdd90454-02a8-405a-970c-66b3f0ea2c96.png">